src/AreaView/AreaCard/AreaCard.tsx

```diff
@@ -1,48 +1,79 @@
 import * as React from "react";
 import { Area } from "../../Area/Area";
 import { Card } from "azure-devops-ui/Card";
-import { Splitter, SplitterElementPosition } from "azure-devops-ui/Splitter";
 import { IPeoplePickerProvider } from "azure-devops-ui/IdentityPicker";
 import "../AreaView.scss";
 import { AreaCardIdentity } from "./AreaCardIdentity";
 import { AreaCardDetails } from "./Details/AreaCardDetails";
 import { Dialog } from "azure-devops-ui/Dialog";
+import { useStateValue } from "../../StateMangement/StateProvider";
+import produce from "immer";
+import { AreaCardButtons } from "./AreaCardButtons";
 
 export interface IAreaCardProps {
-    area: Area;    
+    area: Area;
     identityProvider: IPeoplePickerProvider;
     removeAreaCallback: (id: string, areaId: string) => void;
     onCardClick: (area: Area) => void;
 }
 
 export const AreaCard: React.FunctionComponent<IAreaCardProps> = props => {
     const { area, identityProvider, removeAreaCallback, onCardClick } = props;
-    const [{ editMode, isDialogOpen }, localDispatcher] = React.useState({ editMode: false, isDialogOpen: false });
+    const [{ editMode, isDialogOpen, draftArea }, localDispatcher] = React.useState({ editMode: false, isDialogOpen: false, draftArea: undefined });
+    const stateContext = useStateValue();
 
     const toggleEditMode = () => {
-        localDispatcher({ editMode: !editMode, isDialogOpen: isDialogOpen });
-    };
+        // If starting to edit, set the draft to the passed in area. 
+        // If exiting editing, clear the draft. 
+        const exitingEditMode = editMode;
```

Comment:
@Nianwens When we toggle the edit state, I either initialize or clear the draft area. Below, I pass the correct area to the child components. 

The only negative I see here is if we reverse the order of editArea and toggleEditMode in the `save()` method (line 66), it will break. Does this signal the solution is fragile. Everyone, let me know if you have suggestions!

Reply:
I think your comment is sufficient. I also think it's intuitive to assume that the order for updating state and UI is important (i.e. if the update fails, we shouldn't continue the UI flow in an unstable state), thus someone shouldn't mess with it arbitrarily anyway..

Reply:
Agree with Mike, the order is important, that's also another reason React can benefit us,  if you are interested, we can brainstorm whether there are other solutions today as an exercise :).